### PR TITLE
Aliases subpage

### LINKED ISSUE #16: Admin subpage: Aliases
Definition of done:
 - no styling, plain html
 - table with 4 columns: id (from 0 up), peerId, alias, actions
 - import and export (into csv file) buttons
 - add alias button
 - remove alias button (in actions of the table)
 - refresh button needed at the same line as title of the page, button should be on the right

### PR DESCRIPTION
**Pull request by diegoalzate**
closes #16 

### overview
provides aliases interaction like deleting/creating aliases. this pr contains no specific styles, the table is created with mui so it definitely looks better than the rest of things.

#### things to know moving forward
- created a component to import csv data that can be re used
- created a export to csv function that can also be re used

### CODE REVIEW

File: src/sections/aliases.tsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+    // Reset the file input value
+    if (fileInputRef.current) {
+      fileInputRef.current.value = "";
+    }
+  };
+
+  return (
+    <div>
+      import <input type="file" accept=".csv" ref={fileInputRef} onChange={handleFileUpload} placeholder="import" />
```
Comment (mjadach-iv) [inline src/sections/aliases.tsx:205]: Here just add an 'Import' button which will open the File Explorer for importing.
All that without the import text and 'no file chosen' etc.

File: src/sections/aliases.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    setForm({ ...form, [name]: value })
+  }
+
+  return (
+    <div>
+      <input type="text" name="peerId" placeholder="peerId" onChange={handleChange} value={form.peerId} />
+      <input type="text" name="alias" placeholder="alias" onChange={handleChange} value={form.alias} />
+      <button onClick={() => {
+        if (loginData.ip && loginData.apiKey) {
+          dispatch(actionsAsync.setAliasThunk({ alias: form.alias, peerId: form.peerId, apiEndpoint: loginData.ip, apiToken: loginData.apiKey }))
```
Comment (mjadach-iv) [inline src/sections/aliases.tsx:110]: If that call will fail, we need an error to show up under the form with information what went wrong.
If the call will succeed, we show information under the form that it;'s saved and we clear the textfield.

File: src/sections/aliases.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            {Object.entries(aliases ?? {}).map(([alias, peerId], key) => (
+              <TableRow
+                key={key}
+              >
+                <TableCell component="th" scope="row">
+                  {key}
+                </TableCell>
+                <TableCell>{peerId}</TableCell>
+                <TableCell>{alias}</TableCell>
+                <TableCell><DeleteAliasButton alias={alias} /></TableCell>
```
Comment (mjadach-iv) [inline src/sections/aliases.tsx:69]: We need a state in the component in which we will store 'table action errors'.
If this call fails, save the message to this state.
In next PRs we will add Notification center or information snackbar center.

File: src/sections/aliases.tsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+  return (
+    <Section
+      className="Section--aliases"
+      id="Section--aliases"
+      yellow
+    >
+      <h2>Add new alias</h2>
+      <CreateAliasForm />
+      <h2>Aliases table</h2>
```
Comment (mjadach-iv) [inline src/sections/aliases.tsx:28]: in this line, please also add a refresh button, which will download aliases again

**Review by mjadach-iv — CHANGES_REQUESTED** [review]

File: src/sections/aliases.tsx (hunk 1 above)
Comment (diegoalzate) [reply src/sections/aliases.tsx:205]: I can do this by hidding the <input /> and triggering it through a button but not sure how a button can open the file explorer without the <input />

**Review by ItsLuized — APPROVED** [review]

File: src/sections/aliases.tsx [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loginData]);
+
+  return (
+    <Section className="Section--aliases" id="Section--aliases" yellow>
+      <h2>Add new alias</h2>
+      <CreateAliasForm />
+      <h2>Aliases table</h2>
```
Comment (mjadach-iv) [inline src/sections/aliases.tsx:42]: Under the table please add the same error handling as you added in the Add new alias

**Review by mjadach-iv — CHANGES_REQUESTED** [review]

**Review by mjadach-iv — APPROVED** [review]

**Pull request merged**